apps/web-giddh/src/app/new-inventory/component/bulk-stock-advance-filter/bulk-stock-advance-filter.component.ts

```diff
@@ -0,0 +1,130 @@
+import { ChangeDetectionStrategy, Component, Input, OnInit, Output } from '@angular/core';
+import { EventEmitter } from '@angular/core';
+@Component({
+    selector: 'bulk-stock-advance-filter',
+    templateUrl: './bulk-stock-advance-filter.component.html',
+    styleUrls: ['./bulk-stock-advance-filter.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class BulkStockAdvanceFilterComponent implements OnInit {
+
+    @Input() public commonLocaleData: any = {};
+    @Input() public localeData: any = {};
+    @Output() public applyAdvanceSearchEvent: EventEmitter<string> = new EventEmitter();
+    @Output() public closeDailog: EventEmitter<string> = new EventEmitter();
+
+    /* Hold advance search SortBy   */
+    public advanceSearchSortBy: any[] = [];
+    /* Hold advance search SortBy options*/
+    public advanceSearchSortByOptions: any[] = [];
+    /* Hold Advance Search Expression*/
+    public advanceSearchExpression: any[] = [];
+    /** Instance of Advance Search Form*/
+    public advanceSearchFormObj: any;
+    constructor() { }
+
+    public ngOnInit(): void {
+        this.formInit();
+        this.formObjectIntialvalue();
+    }
+    /**
```

Comment:
use proper format for all method documentation using document this extension of vs code